src/Parallel/NodeLock.hpp

```diff
@@ -40,10 +42,12 @@ class NodeLock {
 
   void destroy() noexcept;
 
+  bool is_destroyed() noexcept { return nullptr == lock_; }
+
+  ///
```

Comment:
A good start to the comment... Though I don't know if the `pup` function is what we need to document :)